app/scripts/model/dataProxy.js

```diff
@@ -1279,6 +1280,94 @@ window.DataManagerForIviz = (function($, _) {
       getClinicalData: function(attribute_ids, isPatientAttributes) {
         return isPatientAttributes ? this.getPatientClinicalData(attribute_ids) :
           this.getSampleClinicalData(attribute_ids);
+      },
+      getAllGenePanelSampleIds: window.cbio.util.makeCachedPromiseFunction(
+        function (self, fetch_promise) {
+          var _map = {};
+          var asyncAjaxCalls = [];
+          var responses = [];
+          _.each(self.getCancerStudyIds(), function(_studyId) {
+            asyncAjaxCalls.push(
+              $.ajax({
+                url: window.cbioURL + 'api-legacy/genepanel/data',
+                contentType: "application/json",
+                data: ["profile_id=" + _studyId + "_mutations", "genes="].join("&"),
```

Comment:
I think gene list is mandatory for api-legacy/genepanel/data' api. If no gene is passed then the response would be an empty list(object)